geode-core/src/main/java/org/apache/geode/internal/cache/GemFireCacheImpl.java

```diff
@@ -5294,4 +5294,13 @@ public void invokeRegionEntrySynchronizationListenersAfterSynchronization(
       }
     }
   }
+
+  @Override
+  public Object convertPdxInstanceIfNeeded(Object obj) {
```

Comment:
Can this be moved to region...The callers are all from region operation.

Reply:
I put it on Cache because it is the component that knows how to do this conversion.
A Region does not know how to do this. All the pdx config, includes pdx-read-serialized, currently belongs to the cache.